src/index.js

```diff
@@ -95,48 +111,195 @@ class Masq {
     })
   }
 
+  /**
+   * If this is the first time, this.dbs.profiles is empty.
+   * We need to get masq-profiles hyperdb key of masq.
+   */
+
   requestMasqAccess () {
     // Subscribe to channel for a limited time to sync with masq
-    this.hub = signalhub(this.channel, ['localhost:8080'])
+    const hub = signalhub(this.channel, [HUB_URL])
+    let sw = null
+
+    if (swarm.WEBRTC_SUPPORT) {
+      sw = swarm(hub)
+    } else {
+      sw = swarm(hub, { wrtc: require('wrtc') })
+    }
+
+    sw.on('peer', (peer, id) => {
+      peer.on('data', data => _handleData(data, peer))
+    })
+
+    sw.on('close', () => {
+      hub.close()
+    })
+
+    sw.on('disconnect', (peer, id) => {
+      sw.close()
+      hub.close()
+    })
+
+    const _handleData = async (data, peer) => {
+      const json = JSON.parse(data)
+
+      switch (json.msg) {
+        case 'sendProfilesKey':
+          // check challenges
+          if (json.challenge !== this.challenge) {
+            // This peer may be malicious, close the connection
+            sw.close()
+            hub.close()
+          } else {
+            // db creation
+            const db = hyperdb(rai('masq-profiles'), Buffer.from(json.key, 'hex'), { valueEncoding: 'json' })
+            await dbReady(db)
+
+            // Store
+            this.dbs.profiles = db
+
+            peer.send(JSON.stringify({
+              msg: 'replicationProfilesStarted'
+            }))
+            // db replication
```

Comment:
If everything is ok, we should close the hub and sw here right ?

Reply:
yes, added ws.close()